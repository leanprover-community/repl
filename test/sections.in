{"cmd": "section\nopen Nat\nend", "namespaces": true}

{"cmd": "open Lean\nsection test\nopen Nat\nend test", "namespaces": true}

{"cmd": "open Lean\nsection\nopen Nat\nsection test\nopen Elab\nend test\nend", "namespaces": true}
