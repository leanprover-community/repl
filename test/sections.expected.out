{"namespaces":
 [{"pp": "section",
   "pos": {"line": 1, "column": 0},
   "openDecls": [],
   "endPos": {"line": 1, "column": 7},
   "currentNamespace": "[anonymous]"},
  {"pp": "open Nat",
   "pos": {"line": 2, "column": 0},
   "openDecls": ["Nat"],
   "endPos": {"line": 2, "column": 8},
   "currentNamespace": "[anonymous]"},
  {"pp": "end",
   "pos": {"line": 3, "column": 0},
   "openDecls": [],
   "endPos": {"line": 3, "column": 3},
   "currentNamespace": "[anonymous]"}],
 "env": 0}

{"namespaces":
 [{"pp": "open Lean",
   "pos": {"line": 1, "column": 0},
   "openDecls": ["Lean"],
   "endPos": {"line": 1, "column": 9},
   "currentNamespace": "[anonymous]"},
  {"pp": "section test",
   "pos": {"line": 2, "column": 0},
   "openDecls": ["Lean"],
   "endPos": {"line": 2, "column": 12},
   "currentNamespace": "[anonymous]"},
  {"pp": "open Nat",
   "pos": {"line": 3, "column": 0},
   "openDecls": ["Nat", "Lean"],
   "endPos": {"line": 3, "column": 8},
   "currentNamespace": "[anonymous]"},
  {"pp": "end test",
   "pos": {"line": 4, "column": 0},
   "openDecls": ["Lean"],
   "endPos": {"line": 4, "column": 8},
   "currentNamespace": "[anonymous]"}],
 "env": 1}

{"namespaces":
 [{"pp": "open Lean",
   "pos": {"line": 1, "column": 0},
   "openDecls": ["Lean"],
   "endPos": {"line": 1, "column": 9},
   "currentNamespace": "[anonymous]"},
  {"pp": "section",
   "pos": {"line": 2, "column": 0},
   "openDecls": ["Lean"],
   "endPos": {"line": 2, "column": 7},
   "currentNamespace": "[anonymous]"},
  {"pp": "open Nat",
   "pos": {"line": 3, "column": 0},
   "openDecls": ["Nat", "Lean"],
   "endPos": {"line": 3, "column": 8},
   "currentNamespace": "[anonymous]"},
  {"pp": "section test",
   "pos": {"line": 4, "column": 0},
   "openDecls": ["Nat", "Lean"],
   "endPos": {"line": 4, "column": 12},
   "currentNamespace": "[anonymous]"},
  {"pp": "open Elab",
   "pos": {"line": 5, "column": 0},
   "openDecls": ["Lean.Elab", "Nat", "Lean"],
   "endPos": {"line": 5, "column": 9},
   "currentNamespace": "[anonymous]"},
  {"pp": "end test",
   "pos": {"line": 6, "column": 0},
   "openDecls": ["Nat", "Lean"],
   "endPos": {"line": 6, "column": 8},
   "currentNamespace": "[anonymous]"},
  {"pp": "end",
   "pos": {"line": 7, "column": 0},
   "openDecls": ["Lean"],
   "endPos": {"line": 7, "column": 3},
   "currentNamespace": "[anonymous]"}],
 "env": 2}

